sumo/src/main.rs

```diff
@@ -0,0 +1,193 @@
+//! Converts a SUMO .net.xml into an A/B Street map.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use abstutil::{CmdArgs, MapName, Tags, Timer};
+use geom::{Distance, PolyLine};
+use map_model::{
+    osm, raw, AccessRestrictions, Intersection, IntersectionID, IntersectionType, Lane, LaneID,
+    LaneType, Map, Road, RoadID, Turn, TurnID, TurnType,
+};
+
+use sumo::{Direction, Network, NodeID};
+
+fn main() {
+    let mut timer = Timer::new("convert SUMO network");
+    let mut args = CmdArgs::new();
+    let input = args.required_free();
+    args.done();
+
+    let network = Network::load(&input, &mut timer).unwrap();
+    let map = convert(&input, network);
+    map.save();
+}
+
+fn convert(orig_path: &str, network: Network) -> Map {
+    let mut intersections = Vec::new();
+    let mut ids_intersections: BTreeMap<NodeID, IntersectionID> = BTreeMap::new();
+    for (_, junction) in network.junctions {
+        let id = IntersectionID(intersections.len());
+        intersections.push(Intersection {
+            id,
+            polygon: junction.shape,
+            turns: BTreeSet::new(),
+            elevation: Distance::ZERO,
+            intersection_type: IntersectionType::StopSign,
+            orig_id: osm::NodeID(123),
+            incoming_lanes: Vec::new(),
+            outgoing_lanes: Vec::new(),
+            roads: BTreeSet::new(),
+        });
+        ids_intersections.insert(junction.id, id);
+    }
+    let mut roads: Vec<Road> = Vec::new();
+    let mut lanes = Vec::new();
+    let mut ids_lanes: BTreeMap<sumo::LaneID, LaneID> = BTreeMap::new();
+    for (_, edge) in network.normal_edges {
+        let src_i = ids_intersections[&edge.from];
+        let dst_i = ids_intersections[&edge.to];
+        // SUMO has one edge in each direction, but ABST has bidirectional roads. Detect if this
+        // edge is the reverse of one we've already handled.
+        let (road_id, direction) =
+            if let Some(r) = roads.iter().find(|r| r.dst_i == src_i && r.src_i == dst_i) {
+                (r.id, map_model::Direction::Back)
+            } else {
+                (RoadID(roads.len()), map_model::Direction::Fwd)
+            };
+
+        let mut lanes_rtl: Vec<(LaneID, map_model::Direction, LaneType)> = Vec::new();
+        for lane in &edge.lanes {
+            let lane_id = LaneID(lanes.len());
+            ids_lanes.insert(lane.id.clone(), lane_id);
+            let lane_type = if lane.allow == vec!["pedestrian".to_string()] {
+                LaneType::Sidewalk
+            } else if lane.allow == vec!["bicycle".to_string()] {
+                LaneType::Biking
+            } else if lane.allow == vec!["rail_urban".to_string()] {
+                LaneType::LightRail
+            } else {
+                LaneType::Driving
```

Comment:
Are there multiple String values that would map to LaneType::Driving? 

If not, you might catch a logic error by adding something like `debug_assert!(lane.allow == "driving")`

Reply:
There are many other cases here, so for now, I'm eyeballing the result to make sure this is reasonable. For example, a residential OSM way maps over to `disallow="tram rail_urban rail rail_electric rail_fast ship"`